### ISSUE TITLE
Implement replay gain

### ISSUE THREAD

**Issue by DJCALIEND** [aidan-g/FoxTunes#58]
Too much distortion between song. I have to level up/down volume for each song (well almost).

**Comment by aidan-g** [comment]
Blocked by #56 for now.

**Comment by aidan-g** [comment]
Creating replay gain tags is complicated but we could implement the playback part (once we have a volume control) by respecting the ReplayGainTrackGain and ReplayGainAlbumGain tags.

**Comment by aidan-g** [comment]
Please post some tracks with replay gain tags.

**Comment by aidan-g** [comment]
Enable the replay gain options;
![ReplayGain_1](https://user-images.githubusercontent.com/13859177/78944344-98d53400-7ab5-11ea-83d1-65b7a6a6e514.PNG)
And check the status by hovering over the settings button during playback;
![ReplayGain_2](https://user-images.githubusercontent.com/13859177/78944392-b3a7a880-7ab5-11ea-8561-2c8d94767f2d.PNG)
Here is a build to test;
[FoxTunes-2020-04-09-nightly-net461.zip](https://github.com/aidan-g/FoxTunes/files/4458847/FoxTunes-2020-04-09-nightly-net461.zip)

**Comment by DJCALIEND** [comment]
Seems to be perfect. I need more tests but it works. Many thanks :)

**Comment by aidan-g** [comment]
I think we need to work on the math a little. 
We need to covert dB as the replay gain info is in decibels to device power 0 to 1 or 0% mute to 100% full volume. 

I’m using dB = replay gain, volume = pow(10, dB) / 20

I’m not very good at maths :(

**Comment by DJCALIEND** [comment]
Humm I think we have a regression with cover art in this test build. Only the first played song show me cover and after it's the same cover. Your replay gain works fine (It's clearly better than before). I am testing.

**Comment by aidan-g** [comment]
It sounds like 87f5cf89e88931480f54d40ebce31d9cc3860035.
Do your tracks have "Album" tags? If not it will use one image per folder.

This problem is actually quite difficult, to create a unique identifier for the artwork. 

It sounds like artist + year + album should work but then you get multiple images for compilations as each track has a different artist *but* it's really the same album and should only have a single image.

I think falling back to the image checksum will fix your problem;

```diff
         private async Task<string> ImportImage(File file, Tag tag, IPicture picture, ArtworkType type, bool overwrite)
         {
-            var id = this.GetPictureId(file, tag, type);
+            var id = this.GetPictureId(file, tag, picture, type);
             return await this.ImportImage(picture, id, overwrite).ConfigureAwait(false);
         }
 
@@ -609,10 +609,10 @@ namespace FoxTunes
             Logger.Write(this, LogLevel.Error, "Failed to read meta data: {0}", e.Message);
         }
 
-        private string GetPictureId(File file, Tag tag, ArtworkType type)
+        private string GetPictureId(File file, Tag tag, IPicture picture, ArtworkType type)
         {
             //Year + (Album | Directory) + Type
-            var hashCode = default(int);
+            var hashCode = default(long);
             unchecked
             {
                 if (tag.Year != 0)
@@ -625,7 +625,7 @@ namespace FoxTunes
                 }
                 else
                 {
-                    hashCode += global::System.IO.Path.GetDirectoryName(file.Name).ToLower().GetHashCode();
+                    hashCode += picture.Data.Checksum;
                 }
                 hashCode = (hashCode * 29) + type.GetHashCode();
             }
```

What do you think?

**Comment by DJCALIEND** [comment]
I don't have "Album" tag, as is not useful for me. I only have (for 3/4 of my songs) "single" songs (like vinyl or "single CD").

**Comment by aidan-g** [comment]
Is this better?
[FoxTunes-2020-04-10-nightly-net461.zip](https://github.com/aidan-g/FoxTunes/files/4461439/FoxTunes-2020-04-10-nightly-net461.zip)

**Comment by DJCALIEND** [comment]
Perfect :) +1. Many thanks.

**Comment by aidan-g** [comment]
If you find any information on how exactly replay gain should be calculated let me know as this is probably wrong.

```C#
public static float GetVolume(float replayGain)
{
    //Convert gain in dB to power 0..1.
    return Convert.ToSingle(Math.Pow(10, replayGain / 20));
}
```

**Comment by aidan-g** [comment]
Implemented since 265864b212a032a0ee7d57d9fc07a25f41bb13e3
